### PR TITLE
Portal bug fixes and features update

### LINKED ISSUE #7599: Portal updates and bug fixes
### Describe the bug

I'm doing a complete portal review with focus on bugs I'll list in PR as I find and fix.
Also looking to refactor secure messaging to either twig or PHP template but mission is to get rid of angularjs.
May do some structural changes to continue portal refactor from a single page app. Staging for module support.
Review and fix support for document audits workflow. Still need a way for audit reviewer to send document back to patient if changes are needed.
While email notification for patient to address a portal form is now allowed, it requires faxsms module turned on for sending in Dashboard. Since I created a new email client module I need to allow email turned on as separate use in SMS dialogs.

Add test that an email can be sent through openemr config'd SMTP before allowing registration to continue. 
Will try to do after patients email is verified as valid. This means I'll have to send patient an email with a redirect with auth token back to registration. A lot of work for sure.

The portal is important to me and want to see it finish with document templates complete integration.

This issue will cover several PR's

### Client configuration

- Browser:
- OpenEMR version:
- Operating system:

### PR DESCRIPTION
**Pull request by sjpadgett**
- missed validation for username length change.
- PHP warnings

Fixes #7599

### CODE REVIEW

File: portal/index.php [hunk 1]
```diff
@@ -320,7 +320,7 @@
         function checkUserName() {
             let vacct = document.getElementById('uname').value;
             let vsuname = document.getElementById('login_uname').value;
-            if (vsuname.length < 12) {
+            if (vsuname.length < 8) {
                 alert(<?php echo xlj('User Name must be at least 12 characters!'); ?>);
```
Comment (bradymiller) [inline portal/index.php:324]: would change above message to ...at least 8 characters...

**Comment by sjpadgett** [comment]
registration has been testing well.

File: templates/portal/registration/portal_register.html.twig [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <title>{{ 'New Patient'|xlt }} | {{ 'Register'|xlt }}</title>
+    <meta name="description" content="Developed By sjpadgett@gmail.com" />
+
+    {{ setupHeader(['no_main-theme', 'portal-theme', 'datetime-picker']) }}
+
+    <script>
+        const webRoot = {{ global.web_root|json_encode|raw }};
```
Comment (bradymiller) [inline templates/portal/registration/portal_register.html.twig:24]: wouldn't use raw (we don't use autoescaping, so doesn't do anything)

File: templates/portal/registration/portal_register.html.twig [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+                data: $("#insuranceForm").serialize()
+            }).done(function () {
+                doCredentials(provider);
+            });
+        }
+
+        function handleInsuranceChange() {
+            if ($('#inscompany').val().toUpperCase() === 'SELF') {
+                $("#insuranceForm input").removeAttr("required");
+                alert('{{ "You have chosen to be self insured or currently do not have insurance. Click next to continue registration."|xlj }}');
```
Comment (bradymiller) [inline templates/portal/registration/portal_register.html.twig:166]: remove those outside single quotes (xlj adds them)

File: templates/portal/registration/portal_register.html.twig [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+            }
+        }
+
+        function validateDOB(dobInput) {
+            const bday = dobInput.val() ?? '',
+                age = Math.round(Math.abs((new Date().getTime() - new Date(bday).getTime())) / (1000 * 60 * 60 * 24));
+            if (age < 30) {
+                alert('{{ "Invalid Date format or value! Type date as YYYY-MM-DD or use the calendar."|xlj }}');
```
Comment (bradymiller) [inline templates/portal/registration/portal_register.html.twig:174]: remove outside single quotes

**Pull request merged**